### DISCUSSION TITLE
feat: Add sdk pack to IOS and invoke it's method in main plugin application

### DISCUSSION THREAD

**Discussion by magiclockers** [ionic-team/capacitor#5228] [Q&A]
## Feature Request
need integrate this package into custom plugin main
[iOS SDK Development Manual.docx](https://github.com/ionic-team/capacitor-plugins/files/7491670/iOS.SDK.Development.Manual.docx)
### Plugin

**Custome plugin is working, inside m file of IOS I have**
CAP_PLUGIN(ContactsPlugin, "ContactsPlugin",
           CAP_PLUGIN_METHOD(echo, CAPPluginReturnPromise);
)
**and in swift file I have the function:**
@objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.success([
            "value": value
        ])
    }

**which is perfectly working with plugin, however..**
### Description

I added a sdk package to it, I want to  init The SDK  before invoking its functional interface:
- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
    [AirbnkSDK initAirbnkSDK];
return YES;
}
**and then,Set up the delegate where necessary:**
@interface LockViewController ()<AirbnkSDKDelegate>
@end
@implementation LockViewController
- (void)viewDidLoad {
    [super viewDidLoad];
    [AirbnkSDK handleDelegate:self];
}

### Platform(s)

IOS platform, need to add a function to invoke the sdk package

### Preferred Solution

then I try to add this:
@interface AppDelegate : UIResponder <UIApplicationDelegate>

@property (strong, nonatomic) UIWindow *window;

@end
@interface AppDelegate ()

@end

@implementation AppDelegate

- (BOOL)application:(UIApplication *)application didFinishLaunchingWithOptions:(NSDictionary *)launchOptions {
    [AirbnkSDK initAirbnkSDK:@"airbnkHaha123456"];
    return YES;
}
@end
**to initiate this additional package**

### Alternatives

Can I then invoke method something like this please??
class LockViewController {
    func scanAndConnect() {
        hud = HudUtils.createMBProgressHUD(withStr: "", inView: view)
        AirbnkSDK.connect(toLock: UserDefaults.standard.object(forKey: SNINFO_KEY))

        hud = HudUtils.createMBProgressHUD(withStr: "", inView: view)
        AirbnkSDK.unlock(UserDefaults.standard.object(forKey: SNINFO_KEY))
    }
}

### Additional Context

my goal is to invoke the method from sdk package to the plugin swift file so that call function in main method of the plugin
thanks for any suggests/helps

**Comment by jcesarmobile** [comment]
The first method called on a plugin initialization is the load method, you can try putting any initalization code you need there

example:
https://github.com/ionic-team/capacitor-plugins/blob/main/network/ios/Plugin/NetworkPlugin.swift#L8

**Comment by magiclockers** [comment]
updates for what I've done on this issue:

`import Foundation
import Capacitor


/**
 * Please read the Capacitor iOS Plugin Development Guide
 * here: https://capacitorjs.com/docs/plugins/ios
 */
@objc(ContactsPlugin)
public class ContactsPlugin: CAPPlugin {

    @objc func echo(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        let app = AppDelegate()
        let initiate = app.application
        print(initiate)
        let sn = "TiHeSBp9BlLKFW6jgfondDy9ovkoUGwbsFLES3HYkRPQ48KL5aZe9dMtLCS/I44M/wVZWV0KUqUdJT1oq9FR//8FWVldClKlHSU9aKvRUf//BVlZXQpSpR0lPWir0VH//wVZWV0KUqUdJT1oq9FR//igVOBiVnz7W2LScEbB3TdUWjQH+gTO5sfaexY2pOf+39SoNpk07DhbbEfkoq";
        let v = app.scanAndConnect(sn:sn)
        let c = app.lockerStatus(sn: sn)
        print(c)
        let pass = app.getPass(sn: sn)
        call.success([
            "value": pass
        ])
    }
    @objc func getContacts(_ call: CAPPluginCall) {
        let value = call.getString("value") ?? ""
        call.success([
            "value": value
        ])
    }

}
//import AirbnkSDK.h
@UIApplicationMain
class AppDelegate: UIResponder, UIApplicationDelegate {
    var window: UIWindow?
    var sdk = AirbnkSDK() // AirbnkSDKDelegate()

    func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey : Any]? = nil) -> Bool {
        AirbnkSDK.initAirbnkSDK("airbnb123456789")
        return true
    }
        
    func scanAndConnect(sn:String) -> () {
         AirbnkSDK.connect(toLock: sn)
    }
    func unlock(sn:String) -> (){
        AirbnkSDK.unlock(sn)
    }
    func lockerStatus(sn:String) -> (){
        AirbnkSDK.getLockStatus(sn)
    }
    func getPass(sn:String) -> (){
        AirbnkSDK.getOneTimePassWord(sn)
    }

}`
